lib/CConfigHandler.cpp

```diff
@@ -53,18 +54,26 @@ SettingsStorage::SettingsStorage():
 {
 }
 
-void SettingsStorage::init()
+void SettingsStorage::init(bool p)

```

Comment:
Perhaps provide filenames explicitly by caller? And perform validation only if scemaFilename is not empty?
```suggestion
void SettingsStorage::init(const std::string & dataFilename, const std::string & schemaFilename)

```